include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -970,11 +972,11 @@ def StructCreateOp : MooreOp<"struct_create", [SameOperandsAndResultType]> {
     ```
     See IEEE 1800-2017 § 7.2. "Structures".
   }];
-  let arguments = (ins UnpackedType:$input);
-  let results = (outs UnpackedType:$result);
-  let hasCustomAssemblyFormat = 1;
+  let arguments = (ins StrAttr:$name,  Variadic<UnpackedType>:$input);
```

Comment:
I don't think `struct_create` should have a name. The op creates a new struct SSA value, but that doesn't necessarily mean it has a corresponding Verilog name. The name is more likely to be attached to a `moore.variable`. For the `struct_create` it's okay to just have `(ins Variadic<UnpackedType>:$fields)`.

The `OpAsmOpInterface` then also shouldn't be needed.